### DISCUSSION TITLE
Questions on Overriding Controller Methods and Resource Attributes in Madmin::ResourceController

### DISCUSSION THREAD

**Discussion by oleoneto** [excid3/madmin#28] [Q&A] [answered]
Hi team!

I am rather new to Rails and I have just recently started using this gem for a website I'm building. Loving the design of the gem and how easy it has been to integrate it with my models! Reminds me of how easy it is to get up and running with Django's own built-in admin site.

As I am diving deeper into customizing the site for my use case, I ran into a little bit of blocker regarding some of the abstractions that make the gem work:

I am trying to customize the behavior of a controller, much like how you would a regular controller. In this specific case, I have an articles controller where I'd like to override the author attribute of a new article resource with the current user instead of letting the logged in administrator select a user record from the database.

In a regular rails controller I'd have done something like the following:
```
class ArticlesController < ApplicationController
   def create
       @article.author_id = current_user.id
       # other logic...
       @article.save
   end
end
```

However, there are a few layers of abstraction in the `Madmin::ResourceController` that I am yet to understand. I am experimenting with setting the `author_id` property of either `resource`, `model`, or `record` but doing so raises exceptions about the attribute not being present. I believe I am missing some understanding of how the `ActiveModel` instance is treated and passed around such that it can at times act as a generic model while at other times a concrete model type. Perhaps this assumption is also incorrect.

**So, all of that to say: how could I go about implementing such a feature?**

**Comment by excid3** [comment, marked as answer]
It's pretty straightforward! When you generate a resource, it will generate a matching controller that you can override as needed.

For example `rails g madmin:resource Post` generates `app/controllers/madmin/posts_controller.rb`. You can override the create action there.

Refer to the resources_controller to see what you have accessible and how the normal `create` action looks: https://github.com/excid3/madmin/blob/master/app/controllers/madmin/resource_controller.rb

**Reply by oleoneto** [reply]
Thanks, @excid3! This did it for me:

```
    def create
      @record = resource.model.new(resource_params)
      @record.author = current_user  # <-------------- Author override
      if @record.save
        redirect_to resource.show_path(@record)
      else
        render :new, status: :unprocessable_entity
      end
    end
```
